src/libYARP_OS/include/yarp/os/Port.h

```diff
@@ -258,6 +265,7 @@ class YARP_OS_API Port : public UnbufferedContactable
 private:
     void* implementation;
     bool owned;
+    ReaderDataProcessor dataProcessor;
```

Comment:
This and the `ReaderDataProcessor` class should be hidden inside the implementation,